src/layers/icmpv6.rs

```diff
@@ -0,0 +1,159 @@
+}
+
+/// Structure representing the ICMPv6 Header
+#[derive(Default, Debug, Serialize)]
+pub struct ICMPv6 {
+    #[serde(rename = "type")]
+    icmp_type: u8,
+    code: u8,
+    #[serde(serialize_with = "crate::types::hex::serialize_lower_hex_u16")]
+    checksum: u16,
+    #[serde(flatten)]
+    rest_of_header: Icmpv6Type,
+}
+
+impl ICMPv6 {
+    pub(crate) fn creator() -> Box<dyn Layer + Send> {
+        Box::<ICMPv6>::default()
+    }
+}
+
+impl Layer for ICMPv6 {
+    fn decode_bytes(
+        &mut self,
+        bytes: &[u8],
+    ) -> Result<(Option<Box<dyn Layer + Send>>, usize), Error> {
+        let decoded;
+
+        if bytes.len() < ICMPV6_HEADER_LENGTH {
+            return Err(Error::TooShort {
+                required: ICMPV6_HEADER_LENGTH,
+                available: bytes.len(),
+                data: hex::encode(bytes),
+            });
+        }
+
+        // decode type, code and checksum
+        self.icmp_type = u8::from_be(bytes[0]);
+        self.code = u8::from_be(bytes[1]);
+        self.checksum = (bytes[2] as u16) << 8 | (bytes[3] as u16);
+        self.rest_of_header = match self.icmp_type {
+            ICMPV6_ECHO_REQUEST => {
+                let identifier = (bytes[4] as u16) << 8 | (bytes[5] as u16);
+                let sequence_number = (bytes[6] as u16) << 8 | (bytes[7] as u16);
+                decoded = 8;
+                Icmpv6Type::EchoReply(Icmpv6Echo {
+                    identifier,
+                    sequence_number,
+                })
+            }
+
+            ICMPV6_ECHO_REPLY => {
+                let identifier = (bytes[4] as u16) << 8 | (bytes[5] as u16);
+                let sequence_number = (bytes[6] as u16) << 8 | (bytes[7] as u16);
+                decoded = 8;
+                Icmpv6Type::EchoReply(Icmpv6Echo {
+                    identifier,
+                    sequence_number,
+                })
+            }
+
+            ICMPV6_PACKET_TOO_BIG => {
+                let mtu = u32::from_be_bytes(bytes[4..8].try_into().unwrap());
+                decoded = 8;
+                Icmpv6Type::PacketSizeTooBig(Icmpv6PacketSizeTooBig { mtu })
+            }
+
+            ICMPV6_DESTINATION_UNREACHABLE | ICMPV6_TIME_EXCEEDED => {
+                decoded = 8;
+                Icmpv6Type::Empty
+            }
+
+            ICMPV6_NEIGHBOR_SOLICITATION => {
+                let target_address = bytes[8..24].try_into().unwrap();
+                decoded = 24;
+                Icmpv6Type::NeighborSolicitation(Icmpv6NeighborSolicitation { target_address })
+            }
+
+            _ => {
+                decoded = bytes.len();
+                Icmpv6Type::Unsupported(Icmpv6Unsupported {
+                    unsupported: bytes[4..].to_vec(),
+                })
+            }
+        };
+        Ok((None, decoded))
+    }
+    fn name(&self) -> &'static str {
+        "ICMPV6"
+    }
+
+    fn short_name(&self) -> &'static str {
+        "icmpv6"
+    }
+}
```

Comment:
You may want to add test cases for - 

1. Supported Packet Types. 
2. One  Unsupported Type.